src/main/webapp/app/localci/build-queue/build-queue.component.ts

```diff
@@ -12,6 +12,88 @@ import { onError } from 'app/shared/util/global.utils';
 import { HttpErrorResponse, HttpHeaders, HttpResponse } from '@angular/common/http';
 import { AlertService } from 'app/core/util/alert.service';
 import dayjs from 'dayjs/esm';
+import { NgbModal, NgbTypeahead } from '@ng-bootstrap/ng-bootstrap';
+import { HttpParams } from '@angular/common/http';
+import { LocalStorageService } from 'ngx-webstorage';
+import { Observable, OperatorFunction, Subject, merge } from 'rxjs';
+import { debounceTime, distinctUntilChanged, map, switchMap, tap } from 'rxjs/operators';
+import { UI_RELOAD_TIME } from 'app/shared/constants/exercise-exam-constants';
+
+export class FinishedBuildJobFilter {
+    status?: string = undefined;
+    buildAgentAddress?: string = undefined;
+    buildStartDateFilterFrom?: dayjs.Dayjs = undefined;
+    buildStartDateFilterTo?: dayjs.Dayjs = undefined;
+    buildDurationFilterLowerBound?: number = undefined;
+    buildDurationFilterUpperBound?: number = undefined;
+
+    /**
+     * Adds the http param options
+     * @param options request options
+     */
+    addHttpParams(options: HttpParams): HttpParams {
+        if (this.status && this.status !== '') {
+            options = options.append('buildStatus', this.status);
+        }
+        if (this.buildAgentAddress && this.buildAgentAddress !== '') {
```

Comment:
`''` is already a falsy value

```suggestion
        if (this.buildAgentAddress) {
```